bundles/core/org.eclipse.smarthome.core.transform/src/main/java/org/eclipse/smarthome/core/transform/internal/profiles/TransformationProfile.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.transform.internal.profiles;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.profiles.ProfileCallback;
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.eclipse.smarthome.core.thing.profiles.StateProfile;
+import org.eclipse.smarthome.core.transform.TransformationException;
+import org.eclipse.smarthome.core.transform.TransformationHelper;
+import org.eclipse.smarthome.core.transform.TransformationService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link StateProfile} which transforms a state value send from the handler to the framework based on the provided
+ * {@link ProfileTypeUID}. The direction from the framework toward the handler is left untouched.
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@NonNullByDefault
+public class TransformationProfile implements StateProfile {
+
+    private static Logger logger = LoggerFactory.getLogger(TransformationProfile.class);
+
+    public static final String PROFILE_NAME = "Transformation";
+    private final ProfileTypeUID profileTypeUID;
+    private final TransformationService service;
+
+    private static String PATTERN_PARAM = "pattern";
+
+    @NonNullByDefault({})
+    private final String function;
+    @NonNullByDefault({})
+    private final String format;
+
+    private final ProfileCallback callback;
+
+    public TransformationProfile(ProfileCallback callback, ProfileContext context, TransformationService service,
+            ProfileTypeUID typeUID) {
+
+        this.service = service;
+        this.profileTypeUID = typeUID;
+        this.callback = callback;
+
+        Object paramValue = context.getConfiguration().get(PATTERN_PARAM);
+        logger.debug("Value for parameter '{}' is '{}'", PATTERN_PARAM, paramValue);
+        if (paramValue instanceof String) {
+            String pattern = (String) paramValue;
+            if (pattern.contains(TransformationHelper.FUNCTION_VALUE_DELIMITER)) {
```

Comment:
If you concatenate this condition with the previous one, than you don't need the duplicate else-block and can eliminitate the awkward `logError()`-method.